ipam/allocator.go

```diff
@@ -97,9 +98,9 @@ func NewAllocator(ds datastore.DataStore) (*Allocator, error) {
 
 func (a *Allocator) subnetConfigFromStore(kvPair *store.KVPair) {
 	a.Lock()
-	if a.dbIndex < kvPair.LastIndex {
+	if a.Index() < kvPair.LastIndex {
 		a.subnets = byteArrayToSubnets(kvPair.Value)
```

Comment:
We can't use the the setter functions here ( Index(), SetIndex() ), because of the outer lock otherwise we dead lock.